lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -246,16 +264,20 @@ def self.upload(app_id, binary_path)
               UI.message("Uploading the APK/IPA")
               upload_binary(app_id, binary_path)
             end
-            status = upload_status(upload_token, app_id)
+            status, release_id = upload_status(upload_token, app_id)
           end
         end
+        release_id
       end
 
+      # Gets the upload status for the app release
+      #
+      # Returns the status and the id of the release of applicable
       def self.upload_status(app_token, app_id)
```

Comment:
The name of this method looks outdated now...are you `uploading` a `status` or retrieving the `upload_status` of a thing? Also, now that it returns two variables, should it be updated to be called `get_upload_status_and_release_id`?

Last, I generally try to avoid returning more than one return value from a method as it can get unwieldly pretty quickly.   What combination of status & release_id can be returned?  Does a `release_id` always accompany a status of SUCCESS?  Is it always absent on anything else?  What other status values can be expect?  The definition of the expected results from this backend call might be worth fleshing out and documenting here by creating a more robust response object that encapsulates our expectations.

eg:

class ReleaseUploadStatusResponse
   attr_reader :status, :release_id
   def initialize(status, release_id)
      @status = stats
      @release_id = release_id
   end

   def successful?
      @status == 'SUCCESS'
   end

   # any other logic that might emerge from this refactoring
end
```

Reply:
The method is getting the status of the upload. On success release id has some value so it is not nil. The reason for returning two values is that to get the release id otherwise we would have to make another upload status call once we know it successfully uploaded. The response body from the backend can be seen in the documentation that Tunde made.

Reply:
Based on the doc you shared, I think this API response format might lend itself well to being factored into a simple Ruby object created from the returned JSON (this is adjusted based on the doc you shared):

```

def ReleaseUploadStatusResponse
   def initialize(response_json_hash)
      @response_json_hash = response_json_hash
   end

   def status
      response_json_hash['status']
   end

   def success?
      status == 'SUCCESS' # || maybe other non-failure status count as success too?
   end

   def release_hash
      response_json_hash['release']
   end

   def release_id
      release_hash ? release_hash['id'] : nil
   end
end
 # etc...
```

Then you can use a nice higher level interface without having the client code have to be concerned with the format of the JSON response body.

Reply:
Thank you for the detailed comments and help. I have updated the code to reflect what you commented. Please let me know if I misunderstood anything.

Reply:
Looks great :)  Thanks for taking this on.